src/libraries/System.Formats.Tar/tests/TarReader/TarReader.File.Tests.cs

```diff
@@ -125,28 +125,33 @@ public class TarReader_File_Tests : TarReader_File_Tests_Base
         public void Read_Archive_LongPath_Over255(TarEntryFormat format, TestTarFormat testFormat) =>
             Read_Archive_LongPath_Over255_Internal(format, testFormat);
 
-        [Fact]
-        public void Read_NodeTarArchives_Successfully()
+        [Theory]
+        [InlineData("node-tar")]
+        [InlineData("tar-rs")]
+        public void Read_TestArchives_Successfully(string subset)
```

Comment:
I think this could be improved so that we can investigate failures much more easily.

Instead of iterating through the files inside the test method, you can pass the file names as a MemberData. Plus they can be reused.

```cs
        public static IEnumerable<object[]> GetNodeTarFiles()
        {
            string nodeTarPath = Path.Join(Directory.GetCurrentDirectory(), "tar", "note-tar");
            foreach (string file in Directory.EnumerateFiles(nodeTarPath, "*.tar", SearchOption.AllDirectories))
            {
                yield return new object[] { file };
            }
        }

        public static IEnumerable<object[]> GetTarRsFiles()
        {
            string nodeTarPath = Path.Join(Directory.GetCurrentDirectory(), "tar", "tar-rs");
            foreach (string file in Directory.EnumerateFiles(nodeTarPath, "*.tar", SearchOption.AllDirectories))
            {
                yield return new object[] { file };
            }
        }

     [Theory]
        [MemberData(nameof("GetNodeTarFiles"))]
        [MemberData(nameof("GetTarRsFiles"))]
        public void Read_TestArchives_Successfully(string subset)
```

Then you can remove the forloop inside the test method.